core/src/main/scala/org/apache/spark/h2o/backends/external/ProxyStarter.scala

```diff
@@ -30,9 +30,9 @@ import org.spark_project.jetty.util.thread.{QueuedThreadPool, ScheduledExecutorS
 
 object ProxyStarter extends Logging {
   def startFlowProxy(conf: H2OConf): URI = {
+    var port = findNextFreeFlowPort(conf.clientWebPort, conf.clientBasePort)
     while (true) {
       try {
-        val port = findNextFreeFlowPort(conf)
```

Comment:
we should always start with next port if next iteration if we failed during `server.start()`